kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/ViewResource.kt

```diff
@@ -0,0 +1,30 @@
+/* Licensed under Apache 2.0 (C) 2024 Firezone, Inc. */
+package dev.firezone.android.features.session.ui
+
+import dev.firezone.android.tunnel.model.Resource
+import dev.firezone.android.tunnel.model.Site
+import dev.firezone.android.tunnel.model.StatusEnum
+
+data class ViewResource(
```

Comment:
This was split off from `Resource` because the view doesn't need the `type` field?

Reply:
Might suggest extending `Resource` if possible to avoid duplicating all of these fields.

Reply:
yes, I think in general it's good to split the model of the displayed resource and the resource received from connlib, since they are essentially different.

Reply:
Hm, I'm not sure if I agree with this. I would say they represent the same underlying entity. The whole reason connlib calls onUpdateResources is to send the client something it can work with.

Ideally there would be one model representation, with some fields populated from connlib right?

The reason for this split is only because of serialization limitations with Gson?

Reply:
Of course I'm not primarily responsible for this part of the code base any longer so feel free to take my opinions lightly here :-).

Reply:
i.e. if we were to write tests for this model, would they duplicated?

Reply:
`onUpdateResources` is called to make the client know what resources are there, how they will be represented or even if they will be represented at all in the UI is something that only concerns the UI, so only the UI knows the format of the data that will be most useful to show.

>  i.e. if we were to write tests for these two models, would they duplicated?

These models are just data, what would we test?

Reply:
I think this case, of `disabled` shows that pretty good. it's data only held by the UI, we can artificially plug it into the model sent to us by the callbacks by defaulting to true but then updating it with the true value we hold in memory.

But separating the data types encodes in the type-system that it's a resource where we actually know what `disabled` value is and it's not still expecting to be set, making it less error-prone as to use that value earlier.